src/System.Text.Json/src/System/Text/Json/JsonUtf8Reader_TryGet.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Buffers.Text;
+
+namespace System.Text.Json
+{
+    public ref partial struct JsonUtf8Reader
+    {
+        /// <summary>
+        /// Reads the next JSON token value from the source as a <see cref="string"/>.
+        /// Returns true if the UTF-8 encoded token value can be successfully transcoded
+        /// to a UTF-16 string.
+        /// Returns false if the token type is not JsonTokenType.String or
+        /// JsonTokenType.PropertyName.
+        /// </summary>
+        /// <exception cref="ArgumentException">
+        /// Thrown if invalid byte sequence are detected while transcoding.
+        /// </exception>
+        public bool TryGetValueAsString(out string value)
+        {
+            value = default;
+            if (TokenType != JsonTokenType.String && TokenType != JsonTokenType.PropertyName)
+                return false;
+
+            byte[] valueArray = IsValueMultiSegment ? ValueSequence.ToArray() : ValueSpan.ToArray();
+
+            var utf8 = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
+
+            //TODO: Perform additional validation and unescaping if necessary
+            value = utf8.GetString(valueArray);
+            return true;
+        }
+
+        /// <summary>
+        /// Reads the next JSON token value from the source as a <see cref="double"/>.
+        /// Returns true if the entire UTF-8 encoded token value can be successfully 
+        /// parsed to a double value.
+        /// Returns false if the token type is not JsonTokenType.Number.
+        /// </summary>
+        public bool TryGetValueAsDouble(out double value)
```

Comment:
What does this method do when the precision of the number doesn't fit in double?

For AsnReader my TryGet things are

* Success: Return true
* Right data type, value doesn't fit: Return false
* Wrong data type, you should have checked: Throw.

I like that one because the token type is something the caller can already reason about, and then the return false is about the content.

In this method you're mixing "wrong token type" and "Utf8Parser.TryParse said no", which means I don't know what to do as a caller (the rule for Try methods is all "return false" paths should have the same caller response, and anything other than the one optimized false should throw -- sometimes it feels like "never throw", but there's a lot of nuance)

Reply:
Hmm. Throwing due to incorrect token type is fine by me to avoid overloading the meaning of "false".

> What does this method do when the precision of the number doesn't fit in double?

Returns whatever the `Utf8Parser` is doing, i.e. return false.